src/track/track.cpp

```diff
@@ -1276,6 +1276,39 @@ void Track::importPendingCueInfosMarkDirtyAndUnlock(
     emit cuesUpdated();
 }
 
+QMap<int, MacroPointer> Track::getMacros() const {
+    const auto locked = lockMutex(&m_qMutex);
+    return m_macros;
+}
+
+void Track::setMacros(const QMap<int, MacroPointer>& macros) {
+    const auto locked = lockMutex(&m_qMutex);
+    m_macros = macros;
+}
+
+void Track::addMacro(int slot, const MacroPointer& pMacro) {
+    auto locked = lockMutex(&m_qMutex);
+    m_macros.insert(slot, pMacro);
+    if (pMacro->isDirty()) {
```

Comment:
This check is both useless and wrong. If the insertion succeeded the dirty status of the inserted macro is irrelevant. It is also inconsistent considering the extra logic in isDirty().

Reply:
The extra logic in `isDirty` is only makeshift, as it won't send the `dirty` signal appropriately.
But to get rid of it, I currently only see saving a reference to the Track in the Macro as a solution, which creates a circular reference.

Reply:
You code is wrong and I already explained why. Add a non-dirty macro does not mark the track as dirty as it is supposed to be.

Reply:
oh sorry, I indeed misunderstood you

Reply:
Please take some time time and think about the comments instead of always arguing against and rejecting them in the first place. Otherwise doing reviews is stressful and a waste of time for me.